Simplenote/src/main/java/com/automattic/simplenote/NoteEditorActivity.java

```diff
@@ -235,6 +263,114 @@ public void showTabs() {
         mViewPager.setPagingEnabled(true);
     }
 
+    public void setSearchMatchBarVisible(boolean isVisible) {
+        if (mSearchMatchBar != null) {
+            mSearchMatchBar.setVisibility(isVisible ? View.VISIBLE : View.GONE);
+        }
+    }
+
+    private void setUpSearchMatchBar(Intent intent) {
+        if (mSearchMatchIndexes == null) {
+            String matchOffsets = intent.getStringExtra(NoteEditorFragment.ARG_MATCH_OFFSETS);
+            String[] matches = matchOffsets != null ? matchOffsets.split("\\s+") : new String[]{};
+            String[] matchesStart = new String[matches.length / MATCH_INDEX_COUNT];
+
+            // Get "start" item from matches.  The format is four space-separated integers that
+            // represent the location of the match: "column token start length" ex: "1 3 3 7"
+            for (int i = MATCH_INDEX_START, j = 0; i < matches.length; i += MATCH_INDEX_COUNT, j++) {
+                matchesStart[j] = matches[i];
+            }
+
+            // Remove duplicate items with linked hash set and linked list.
+            matchesStart = new LinkedHashSet<>(new LinkedList<>(Arrays.asList(matchesStart))).toArray(new String[0]);
+            mSearchMatchIndexes = new int[matchesStart.length];
+
+            // Convert matches string array to integer array.
+            for (int i = 0; i < matchesStart.length; i++) {
+                mSearchMatchIndexes[i] = Integer.parseInt(matchesStart[i]);
+            }
+        }
+
+        mSearchMatchBar = findViewById(R.id.search_match_bar);
+        mNumberPosition = findViewById(R.id.text_position);
+        mNumberTotal = findViewById(R.id.text_total);
+        mButtonPrevious = findViewById(R.id.button_previous);
+        mButtonNext = findViewById(R.id.button_next);
+
+        mButtonPrevious.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (mSearchMatchIndex > 0) {
+                    mSearchMatchIndex--;
+                }
+
+                mNoteEditorFragment.scrollToMatch(mSearchMatchIndexes[mSearchMatchIndex]);
+                new Handler().postDelayed(
+                    new Runnable() {
+                        @Override
+                        public void run() {
+                            updateSearchMatchBarStatus();
+                        }
+                    },
+                    getResources().getInteger(android.R.integer.config_mediumAnimTime)
+                );
+            }
+        });
+        mButtonPrevious.setOnLongClickListener(new View.OnLongClickListener() {
```

Comment:
what's the purpose of the long-click? I expected it to go to the first result but here it looks like it's supposed to pop up the search result in a banner on screen? is that the Toast?

Reply:
The long-press shows a hint about the button.  It's a typical feature on Android to long-press on an icon to show text about what action that icon represents since icons can be ambiguous.  It *is* a toast, which is ephemeral lasting two seconds and doesn't block other interface actions.

When opening the note, you *are* scrolled to the first match as you expected.

Reply:
makes so much more sense, thanks for clarifying